creator-node/src/snapbackSM/peerSetManager.js

```diff
@@ -193,14 +203,58 @@ class PeerSetManager {
    * @param {string} endpoint
    * @returns {Object} the /health_check/verbose response
    */
-  async determinePeerHealth (endpoint) {
+  async getPeerHealth (endpoint) {
     // Axios request will throw on timeout or non-200 response
-    return axios({
+    const resp = await axios({
       baseURL: endpoint,
       url: '/health_check/verbose',
       method: 'get',
       timeout: PEER_HEALTH_CHECK_REQUEST_TIMEOUT
     })
+
+    return resp.data.data
+  }
+
+  /**
+   * Takes data off the verbose health check response and determines the peer heatlh
+   * @param {Object} verboseHealthCheckResp verbose health check response
+   *
+   * TODO: consolidate CreatorNodeSelection + peer set health check calculation logic
+   */
+  determinePeerHealth (verboseHealthCheckResp) {
+    const { storagePathSize, storagePathUsed } = verboseHealthCheckResp
+    if (storagePathSize && storagePathUsed && storagePathSize - storagePathUsed <= MINIMUM_STORAGE_PATH_SIZE) {
+      throw new Error(`Almost out of storage=${storagePathSize - storagePathUsed}bytes remaining`)
+    }
+
+    // Check for sufficient memory space of 6GB
+    const { usedMemory, totalMemory } = verboseHealthCheckResp
+    if (usedMemory && totalMemory && totalMemory - usedMemory <= MINIMUM_MEMORY_UNUSED) {
+      throw new Error(`Running low on memory=${totalMemory - usedMemory}bytes remaining`)
+    }
+
+    // Check for sufficient file descriptors space of minimum 5%
+    const { allocatedFileDescriptors, maxFileDescriptors } = verboseHealthCheckResp
+    if (allocatedFileDescriptors && maxFileDescriptors && allocatedFileDescriptors / maxFileDescriptors >= MAX_FILE_DESCRIPTORS_OPEN) {
+      throw new Error(`Running low on file descriptors availability=${allocatedFileDescriptors / maxFileDescriptors * 100}% used`)
+    }
+
+    // Check historical sync data
+    const { dailySyncSuccessCount, dailySyncFailCount } = verboseHealthCheckResp
+    if (dailySyncSuccessCount &&
+      dailySyncFailCount &&
+      dailySyncSuccessCount + dailySyncFailCount > MINIMUM_DAILY_SYNC_COUNT &&
+      dailySyncFailCount / (dailySyncFailCount + dailySyncSuccessCount) > MINIMUM_SYNC_FAIL_COUNT_PERCENTAGE) {
+      throw new Error(`Latest daily sync data shows that this node is not accepting syncs. Successful syncs=${dailySyncSuccessCount} || Failed syncs=${dailySyncFailCount}`)
```

Comment:
nit but this error msga nd the one below on line 256 are a bit misleading, instead of "not accepting syncs" maybe "failing a high rate of syncs" or something